File: app/src/main/java/com/example/jaguar/weather/WeatherActivity.java
package com.example.jaguar.weather;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.List;

public class WeatherActivity extends AppCompatActivity implements WeatherView {
    private WeatherAdapter wAdapter;
    private RecyclerView wRecyclerView;
    private WeatherPresenter weatherPresenter = new WeatherPresenter(this);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_weather);

        wAdapter = new WeatherAdapter(this);
        wRecyclerView = findViewById(R.id.recyclerView);
        wRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        wRecyclerView.setAdapter(wAdapter);

        weatherPresenter.onCreate();

        new UpdateWeather("Москва","ru","7", new CallBack() {
            @Override
            public void UpdateWeather(List<WeatherObjFactory.WeatherObj> objs) {
                    updateWeatherList(objs);
            }
        }).execute();
    }

    private void updateWeatherList(List<WeatherObjFactory.WeatherObj> objs) {
        wAdapter.setAll(objs);
    }
}

File: app/src/main/java/com/example/jaguar/weather/views/WeatherActivity.kt
package com.example.jaguar.weather.views

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.View
import android.widget.AdapterView
import com.arellomobile.mvp.MvpAppCompatActivity
import com.arellomobile.mvp.presenter.InjectPresenter
import com.example.jaguar.weather.R
import com.example.jaguar.weather.adapters.WeatherAdapter
import com.example.jaguar.weather.common.WeatherObj
import com.example.jaguar.weather.presenter.WeatherPresenter
import com.example.jaguar.weather.interfaces.WeatherView
import kotlinx.android.synthetic.main.activity_weather.*
import javax.inject.Inject


class WeatherActivity : MvpAppCompatActivity(), WeatherView {

    private lateinit var wAdapter: WeatherAdapter


    @InjectPresenter
    lateinit var mWeatherPresenter: WeatherPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_weather)

        progressBar.visibility = View.INVISIBLE
        recyclerView.layoutManager = LinearLayoutManager(this)
        wAdapter = WeatherAdapter()
        recyclerView.adapter = wAdapter

        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener{
            override fun onItemSelected(p0: AdapterView<*>?, p1: View?, p2: Int, p3: Long) {
                mWeatherPresenter.onSelect(spinner.selectedItem.toString())
            }
            override fun onNothingSelected(p0: AdapterView<*>?) {
            }
        }
    }
    override fun updateAdapter(objs: MutableList<WeatherObj>) {
        wAdapter.setAll(objs)
    }
    override fun showProgress() {
        progressBar.visibility = View.VISIBLE
    }
    override fun hideProgress() {
        progressBar.visibility = View.INVISIBLE
    }
}